yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -545,8 +545,12 @@ func (yb *TargetYugabyteDB) ExecuteBatch(migrationUUID uuid.UUID, batch *EventBa
 		if err != nil {
 			return false, fmt.Errorf("error creating tx: %w", err)
 		}
-		defer tx.Rollback(ctx)
-
+		defer func() {
+			err = tx.Rollback(ctx)
```

Comment:
BUG: If a `Rollback()` is successful, this code will overwrite the original `err` and return success to the caller (see the named return value of the func passed to `WithConn()`). Use `errRollback := tx.Rollback(ctx)`.